crates/signer/src/wallet/mod.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{Result, Signature, Signer};
```

Comment:
Should wallet be a separate crate?

Reply:
I thought about this, but I don't believe it makes much sense since it uses the same dependencies

Reply:
yeah, i don't love that `Wallet` is special wrt other implementors of `Signer` 🤔  should all of them be `Wallet<T>` ?

Reply:
Don't understand this question, wallet is generic over the underlying private key type basically